### PR TITLE
url constructor javadoc

### LINKED ISSUE #204: *_url constructors can also throw std::length_error
This needs to be documented.

https://github.com/CPPAlliance/url/issues/203#issuecomment-1158852601

### PR DESCRIPTION
**Pull request by alandefreitas**
fix #204

### CODE REVIEW

File: include/boost/url/url.hpp [hunk 1]
```diff
@@ -119,7 +119,15 @@ class BOOST_SYMBOL_VISIBLE url
         relative-ref  = relative-part [ "?" query ] [ "#" fragment ]
         @endcode
 
-        @throw std::invalid_argument parse error.
+        @par Exception Safety
+        Strong guarantee.
+        Exceptions thrown on invalid input.
+        Calls to allocate may throw.
```
Comment (vinniefalco) [inline include/boost/url/url.hpp:125]: I put them in this order
```
        Strong guarantee.
        Calls to allocate may throw.
        Exceptions thrown on invalid input.
```

File: include/boost/url/url.hpp [hunk 2]
```diff
@@ -119,7 +119,15 @@ class BOOST_SYMBOL_VISIBLE url
...
         @endcode
 
-        @throw std::invalid_argument parse error.
+        @par Exception Safety
+        Strong guarantee.
+        Calls to allocate may throw.
+        Exceptions thrown on invalid input.
+
+        @throw system_error
+        The string contains an invalid <em>URI-reference</em>.
```
Comment (vinniefalco) [inline include/boost/url/url.hpp:128]: these exceptions aren't correct

Comment (alandefreitas) [reply include/boost/url/url.hpp:128]: `grammar::parse(s, uri_reference_rule).value()` can throw `system_error` and `url(url_view_base const& u)` can throw `std::length_error`. Is that incorrect?

**Comment by vinniefalco** [comment]
I might have updated these as well, please check my PR

**Pull request closed without merging**